### ISSUE TITLE
Make a real connection pool

### ISSUE THREAD

**Issue by jdegoes** [zio/zio-sql#59]
Currently the `ConnectionPool.live` layer doesn't actually make a connection pool. It should be changed to establish `n` connections with the database, determined by configuration information. Then it should keep track of which ones are in use, and 'lend' them to requestors. 

This could be done in many ways: with STM (`TMap[Connection, Boolean]`), a Queue of connections, etc.

The important part is that connections should be returned to the pool when no longer needed, and the layer itself should shut down all connections as part of the layer's cleanup.

**Comment by phderome** [comment]
I'd suggest that each connection that is created be tested with some trivial SQL statement (or heartbeat if that exists) to the database. If I recall correctly, connection pools typically do this.

**Comment by ChipChopChippy** [comment]
So I'm taking this one

**Comment by jczuchnowski** [comment]
@ChipChopChippy any progress?

**Comment by ChipChopChippy** [comment]
@jczuchnowski Yes - I had a session with @jdegoes on Friday to go through the issue. I'm working on the tests now

**Comment by jczuchnowski** [comment]
@ChipChopChippy awesome. Can't wait to see how you guys approached the problem!
